src/main/kotlin/org/openrewrite/kotlin/internal/PsiElementAssociations.kt

```diff
@@ -152,9 +149,41 @@ class PsiElementAssociations(val typeMapping: KotlinTypeMapping, val file: FirFi
             return typeMapping.type(typeMap[parent], owner)
         }
         val fir = primary(psiElement)
+        if (psiElement != null && fir is FirResolvedQualifier) {
+            if (fir.symbol is FirRegularClassSymbol) {
+                val classId = (fir.symbol as FirRegularClassSymbol).classId
+                return if (isPackage(psiElement, classId)) {
+                    null
+                } else {
+                    val found = matchClassId(if (isOuterMostPsi(psiElement, classId)) psiElement else psiElement.parent, classId)
+                    typeMapping.type(found, owner)
+                }
+            }
+        }
         return if (fir != null) typeMapping.type(fir, owner) else null
     }
 
+    private fun isPackage(psi: PsiElement, classId: ClassId): Boolean {
+        return !classId.packageFqName.isRoot && psi.parent.text == classId.packageFqName.asString()
+    }
+
+    private fun isOuterMostPsi(psi: PsiElement, classId: ClassId): Boolean {
+        val parts = psi.parent.text.split(".")
+        return classId.packageFqName.isRoot && parts.size == 2 && psi.text == parts[0]
```

Comment:
Why it's 2?

Reply:
I'll rename the method and add comments :)